assets/scss/components/_avatar.scss

```diff
@@ -40,37 +33,50 @@ Class                     | Description
-}
+@mixin avatarIconBadge($color) {
+	position: relative;
+	text-indent: initial;
 
-%avatarCharm {
-	overflow: visible;
+	.svg--avatarBadge {
+		position: absolute;
+		bottom: -6%;
+		right: -6%;
+		width: 44%;
+		height: 44%;
+
+		// circle behind the icon
+		&:before {
+			position: absolute;
+			top: 0;
+			right: 0;
+			bottom: 0;
+			left: 0;
+			content: "";
+			background-color: $color;
+			border-radius: 100%;
+		}
+
+		// the icon itself
+		svg {
+			@include color-svg(getPrimaryTextColor($color));
+			width: 80%;
+			height: 80%;
+			transform: translate(12%, 12%);
+		}
+	}
 
-	&:after {
-		background-position: center;
-		background-repeat: no-repeat;
-		background-size: 10px;
-		border-radius: 999px;
+	// Use proportionally smaller badge for larger avatars
+	&.avatar--large .svg--avatarBadge,
+	&.avatar--xxlarge .svg--avatarBadge {
+		width: 30%;
```

Comment:
Is there a reason this is `30%` instead of a more specific value (e.g.: `{n}px`)

Reply:
Yep - using a `%` relative to the parent makes this work with media-scaling avatars